docs/testplan/HA-SmartSwitch-test-plan.md

```diff
@@ -0,0 +1,188 @@
+| Standby T1-T0 link drop - Standby      | Verify packet flow when T1-T0 link drop.                      | • Start  I/O through standby side.<br>• Configure DPU2 side T1-T0 link drop.                      | DPU1 becomes standalone, DPU2 is anything but active. | T2 receives packets without disruption.          |
+
+### Module 6 Critical Process Crash 
+For all process crash cases, we will have 4 variations, it’s 
+1. Process crash on DPU1, traffic landing on DPU2
+2. Process crash on DPU2, traffic landing on DPU1
+3. Process crash on DPU1, traffic landing on DPU1
+4. Process crash on DPU2, traffic landing on DPU2
+The expected behavior is same, that HA state remains unchanged. No data plane disruption is expected. 
+
+| Case            | Goal                                     | Test Steps                                            | Expected Control Plane Behavior                   | Expected Data Plane Behavior                      |
+|-----------------|------------------------------------------|-------------------------------------------------------|--------------------------------------------------|---------------------------------------------------|
+| syncd on DPU    | Verify when syncd crash on DPU.         | • Start  I/O<br>• Kill syncd on DPU       | DPU1 remains active, DPU2 remains standby.        | T2 receives packets without disruption.          |
+| hamgrd on NPU   | Verify when hamgrd crash on NPU.        | • Start  I/O<br>• Kill hamgrd on NPU      | DPU1 remains active, DPU2 remains standby.        | T2 receives packets without disruption.          |
+| pmon on NPU     | Verify when pmon crash on NPU.          | • Start  I/O<br>• Kill pmon on NPU        | DPU1 remains active, DPU2 remains standby.        | T2 receives packets without disruption.          |
+| bgpd on NPU     | Verify when bgpd crash on NPU.          | • Start  I/O<br>• Kill bgpd on NPU        | DPU1 remains active, DPU2 remains standby.        | T2 receives packets without disruption.          |
+
+###  Module 7 Power down and hardware failure
+For each case in this module, there are 2 variations:
+1. Failure happens on DPU1, traffic landing on DPU2
+2. Failure happens on DPU2, traffic landing on DPU1
+
+| Case                   | Goal                                             | Test Steps                                           | Expected Control Plane Behavior                   | Expected Data Plane Behavior                      |
+|------------------------|--------------------------------------------------|------------------------------------------------------|--------------------------------------------------|---------------------------------------------------|
+| DPU hardware failure   | Verify traffic flow when DPU hardware fails     | • Start  I/O<br>• Force DPU reset (ChassisStateDB DPU_STATE) | DPU1 becomes non-active, DPU2 becomes standalone. | T2 receives packets without disruption.          |
```

Comment:
unplanned events will drop packets.

Reply:
Updated for `Module 7 Power down and hardware failure`